confgenerator/logging.go

```diff
@@ -70,6 +72,13 @@ func stackdriverOutputComponent(match, userAgent string, storageLimitSize string
 		// Mute these errors until https://github.com/fluent/fluent-bit/issues/4473 is fixed.
 		"net.connect_timeout_log_error": "False",
 	}
+	r := platform.FromContext(ctx).ResourceOverride
+	if r != nil {
+		if r, ok := r.(resourcedetector.BMSResource); ok {
```

Comment:
The resource type and labels should come from the `ResourceOverride`, instead of being hardcoded here. Maybe that means there should be a second method in the interface (`GCMResource() (string, map[string]string)` or something like that?)

Reply:
Okay, I've created a new method, albeit with a slightly different name. It returns a comma-separated string containing resource labels. This approach simplifies the process by avoiding the need to return a map and then convert it to a comma-separated string while ensuring element order consistency.

I've also update GetType() to return the resource type which is used here in logging.go and in api_check.py.